internal/metrics/listener.go

```diff
@@ -49,5 +53,47 @@ func runServer(port string, registry prometheus.Gatherer) {
 	})
 	address := fmt.Sprintf(":%v", port)
 	glog.Infof("Starting Prometheus listener on: %v%v", address, metricsEndpoint)
-	glog.Fatal("Error in Prometheus listener server: ", http.ListenAndServe(address, nil))
+	if prometheusSecret == nil {
+		glog.Fatal("Error in Prometheus listener server: ", http.ListenAndServe(address, nil))
+	} else {
+		// Unfortunately, http.ListenAndServeTLS() takes a filename instead of cert/key data, so we
+		// Write the cert and key to a temporary file. We create a unique file name to prevent collisions.
+		certFileName := fmt.Sprintf("%v-%s", time.Now().Unix(), "nginx-prometheus.cert")
+		keyFileName := fmt.Sprintf("%v-%s", time.Now().Unix(), "nginx-prometheus.key")
+		certFile, err := writeTempFile(prometheusSecret.Data[api_v1.TLSCertKey], certFileName)
+		if err != nil {
+			glog.Errorf("failed to create cert file for prometheus: %w", err)
```

Comment:
I wonder if it makes sense to Fatal here (and in line 71). Otherwise, the error will occur in `http.ListenAndServeTLS`, which might be less clear that the err here.

Reply:
fixed